internal/pipe/mattermost/mattermost.go

```diff
@@ -0,0 +1,113 @@
+package mattermost
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/apex/log"
+	"github.com/caarlos0/env/v6"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/context"
+)
+
+const (
+	defaultColor           = "#2D313E"
+	defaultUsername        = `GoReleaser`
+	defaultMessageTemplate = `{{ .ProjectName }} {{ .Tag }} is out! Check it out at {{ .GitURL }}/releases/tag/{{ .Tag }}`
+	defaultMessageTitle    = `{{ .ProjectName }} {{ .Tag }} is out!`
+)
+
+type Pipe struct{}
+
+func (Pipe) String() string                 { return "mattermost" }
+func (Pipe) Skip(ctx *context.Context) bool { return !ctx.Config.Announce.Mattermost.Enabled }
+
+type Config struct {
+	Webhook string `env:"MATTERMOST_WEBHOOK,notEmpty"`
+}
+
+func (Pipe) Default(ctx *context.Context) error {
+	if ctx.Config.Announce.Mattermost.MessageTemplate == "" {
+		ctx.Config.Announce.Mattermost.MessageTemplate = defaultMessageTemplate
+	}
+	if ctx.Config.Announce.Mattermost.TitleTemplate == "" {
+		ctx.Config.Announce.Mattermost.TitleTemplate = defaultMessageTitle
+	}
+	if ctx.Config.Announce.Mattermost.Username == "" {
+		ctx.Config.Announce.Mattermost.Username = defaultUsername
+	}
+	if ctx.Config.Announce.Teams.Color == "" {
+		ctx.Config.Announce.Teams.Color = defaultColor
+	}
+
+	return nil
+}
+
+func (Pipe) Announce(ctx *context.Context) error {
+	msg, err := tmpl.New(ctx).Apply(ctx.Config.Announce.Mattermost.MessageTemplate)
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to mattermost: %w", err)
+	}
+
+	title, err := tmpl.New(ctx).Apply(ctx.Config.Announce.Mattermost.TitleTemplate)
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to teams: %w", err)
+	}
+
+	var cfg Config
+	if err := env.Parse(&cfg); err != nil {
+		return fmt.Errorf("announce: failed to announce to mattermost: %w", err)
+	}
+
+	log.Infof("posting: %q", msg)
+
+	wm := &model.IncomingWebhookRequest{
+		Username:    ctx.Config.Announce.Mattermost.Username,
+		IconEmoji:   ctx.Config.Announce.Mattermost.IconEmoji,
+		IconURL:     ctx.Config.Announce.Mattermost.IconURL,
+		ChannelName: ctx.Config.Announce.Mattermost.Channel,
+		Attachments: []*model.SlackAttachment{
+			{
+				Title: title,
+				Text:  msg,
+				Color: ctx.Config.Announce.Teams.Color,
+			},
+		},
+	}
+
+	err = PostWebhook(ctx, cfg.Webhook, wm)
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to mattermost: %w", err)
+	}
+
+	return nil
+}
+
+func PostWebhook(ctx *context.Context, url string, msg *model.IncomingWebhookRequest) error {
```

Comment:
don't need to be exposed:

```suggestion
func postWebhook(ctx *context.Context, url string, msg *model.IncomingWebhookRequest) error {
```

Reply:
thanks again @caarlos0 did the changes you requested